lib/messages/pull-request.js

```diff
@@ -23,7 +24,27 @@ class PullRequest extends AbstractIssue {
 
     this.pullRequest = pullRequest;
     this.statuses = statuses;
+    this.checkRuns = checkRuns;
     this.reviews = reviews;
+
+    this.totalNumberOfStatuses = (statuses || []).length + (checkRuns || []).length;
+
+    this.unsuccessfulStatuses = [];
+    this.successfulStatuses = [];
+
+    if (statuses) {
+      this.unsuccessfulStatuses.push(...statuses.filter(status => status.state !== 'success'));
+      this.successfulStatuses.push(...statuses.filter(status => status.state === 'success'));
+    }
+
+    if (checkRuns) {
+      this.unsuccessfulStatuses.push(...checkRuns.filter(checkRun =>
+        checkRun.conclusion !== 'success' &&
+        checkRun.conclusion !== 'neutral'));
+      this.successfulStatuses.push(...checkRuns.filter(checkRun =>
+        checkRun.conclusion === 'success' ||
+        checkRun.conclusion === 'neutral'));
```

Comment:
We could improve this a little bit if we had the conclusions in an array maybe called `successfulConclusions` and then do `successfulConclusions.includes(checkRun.conclusion)` but it's ok like it is now too 👍